tests/nested/manual/uc-update-assets-secure/task.yaml

```diff
@@ -0,0 +1,122 @@
+summary: Update boot assets through gadgets in secure boot
+
+systems: [ubuntu-20.04-64, ubuntu-22.04-64]
+
+environment:
+  NESTED_ENABLE_SECURE_BOOT: true
+
+  UPDATE_BOOT/both: true
+  UPDATE_SEED/both: true
+
+  UPDATE_BOOT/seed: false
+  UPDATE_SEED/seed: true
+
+  UPDATE_BOOT/boot: true
+  UPDATE_SEED/boot: false
+
+  NESTED_IMAGE_ID: "update-grub-${UPDATE_SEED}-${UPDATE_BOOT}"
+
+prepare: |
+  CHANNEL="$(tests.nested show version)/edge"
+  snap download --basename=pc --channel="${CHANNEL}" pc
+  unsquashfs -d pc pc.snap
+  KEY_NAME=$(tests.nested download snakeoil-key)
+  SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+  SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+
+  # Save the shim before resigning
+  cp pc/shim.efi.signed shim.efi.signed
+
+  # Repack pc gadget for the initial image
+  tests.nested secboot-sign file pc/shim.efi.signed "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  tests.nested secboot-sign file pc/grubx64.efi "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  old_shim_sha="$(sha256sum pc/shim.efi.signed | sed "s/ .*//")"
+  old_grub_sha="$(sha256sum pc/grubx64.efi | sed "s/ .*//")"
+  snap pack pc "$(tests.nested get extra-snaps-path)"
+
+  # Remove signatures
+  cp shim.efi.signed shim.efi
+  while sbverify --list shim.efi | grep "^signature [0-9]*$"; do
+    sbattach --remove shim.efi
+  done
+
+  # Add a different vendor certificate
+  openssl req -new -x509 -newkey rsa:2048 -subj "/CN=new vendor certificate/" -keyout new-cert.key -out new-cert.crt -days 3650 -nodes -sha256
+  openssl x509 -outform der -in new-cert.crt -out new-cert
+  python3 generate_vender_cert_section.py new-section new-cert
+  objcopy --update-section .vendor_cert=new-section shim.efi shim.efi.out
+
+  # Sign modified shim
+  cp shim.efi.out pc/shim.efi.signed
+  tests.nested secboot-sign file pc/shim.efi.signed "${SNAKEOIL_KEY}" "${SNAKEOIL_CERT}"
+
+  if [ "${UPDATE_SEED}" = true ]; then
+    # Resign grub with new vendor key
+    tests.nested secboot-sign file pc/grubx64.efi "new-cert.key" "new-cert.crt"
+  else
+    # If shim is not installed in seed, then we need to keep the snakeoil signature
+    tests.nested secboot-sign file pc/grubx64.efi "${SNAKEOIL_KEY}" "${SNAKEOIL_CERT}"
+  fi
+
+  new_shim_sha="$(sha256sum pc/shim.efi.signed | sed "s/ .*//")"
+  new_grub_sha="$(sha256sum pc/grubx64.efi | sed "s/ .*//")"
+
+  if [ "${UPDATE_SEED}" = true ]; then
+    (cd pc && python3 "${TESTSLIB}/bump_asset_edition.py" --role system-seed)
```

Comment:
the python files could start with
 #!/usr/bin/python3

so then you just execute it in the test. We usually do that in other tests

Reply:
I do not think it is actually very uniform.  For example `tests/lib/core-config.sh` calls `python3 "$TESTSLIB"/manip_seed.py`.

There are also lots of shell scripts in `tests/lib` that have execution permission and never used as such. Only "sourced".